atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java

```diff
@@ -0,0 +1,405 @@
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertLowestFinePartitionInSweepableTimestampsEquals(tsPartitionFine(LOW_TS));
+
+        // after this sweep we progress to sweepTsConservative - 1
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertNoEntriesInSweepableTimestampsBeforeSweepTimestamp();
+    }
+
+    @Test
+    public void sweepableTimestampsGetsScrubbedWhenLastSweptProgressesInNewCoarsePartition2() {
+        long tsSecondPartitionFine = LOW_TS + TS_FINE_GRANULARITY;
+        long largestFirstPartitionCoarse = TS_COARSE_GRANULARITY - 1L;
+        long thirdPartitionCoarse = 2 * TS_COARSE_GRANULARITY;
+        enqueueWrite(TABLE_CONS, LOW_TS);
+        enqueueWrite(TABLE_CONS, tsSecondPartitionFine);
+        enqueueWrite(TABLE_CONS, largestFirstPartitionCoarse);
+        enqueueWrite(TABLE_CONS, thirdPartitionCoarse);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertLowestFinePartitionInSweepableTimestampsEquals(tsPartitionFine(LOW_TS));
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertLowestFinePartitionInSweepableTimestampsEquals(tsPartitionFine(LOW_TS));
+
+        // after this sweep we progress to thirdPartitionCoarse - 1
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertLowestFinePartitionInSweepableTimestampsEquals(tsPartitionFine(thirdPartitionCoarse));
+    }
+
+    @Test
+    public void sweepableCellsGetsScrubbedWheneverLastSweptInNewPartition() {
+        long tsSecondPartitionFine = LOW_TS + TS_FINE_GRANULARITY;
+        enqueueWrite(TABLE_CONS, LOW_TS);
+        enqueueWrite(TABLE_CONS, LOW_TS + 1L);
+        enqueueWrite(TABLE_CONS, tsSecondPartitionFine);
+        enqueueWrite(TABLE_CONS, getSweepTsCons());
+
+        // last swept timestamp: TS_FINE_GRANULARITY - 1
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertSweepableCellsHasEntryForTimestamp(LOW_TS + 1);
+        assertSweepableCellsHasEntryForTimestamp(tsSecondPartitionFine);
+        assertSweepableCellsHasEntryForTimestamp(getSweepTsCons());
+
+        // last swept timestamp: 2 * TS_FINE_GRANULARITY - 1
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertSweepableCellsHasNoEntriesBeforeTimestamp(LOW_TS + 1);
+        assertSweepableCellsHasEntryForTimestamp(tsSecondPartitionFine);
+        assertSweepableCellsHasEntryForTimestamp(getSweepTsCons());
+
+        // last swept timestamp: largestBeforeSweepTs
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertSweepableCellsHasNoEntriesBeforeTimestamp(LOW_TS + 1);
+        assertSweepableCellsHasNoEntriesBeforeTimestamp(tsSecondPartitionFine);
+        assertSweepableCellsHasEntryForTimestamp(getSweepTsCons());
+    }
```

Comment:
Coverage looks solid. I'll see if I can come up with a few more cases, but this looks very nice 🌟